### DISCUSSION TITLE
Rotation around z axis is misbehaving

### DISCUSSION THREAD

**Discussion by waxmar** [sceneview/sceneview#51] [Q&A]
I experienced some strange behaviour using the `ModelNode.rotation` attribute.

Those values work as expected (as long as `y = 0.0f`)
`model.rotation = Rotation(x = 0.0f, y = 0.0f, z = 0.0f)`
car facing towards me with the front

`model.rotation = Rotation(x = 40.0f, y = 0.0f, z = 0.0f)`
front of the car goes down

`model.rotation = Rotation(x = 0.0f, y = 0.0f, z = 40.0f)`
car rotates around z-axis

`model.rotation = Rotation(x = 40.0f, y = 0.0f, z = 40.0f)`
front of car goes down and is rotated around z-axis

Then when a rotation around y-axis is included a rotation around the relative z-axis of the element is no longer working. It seems 
 z values now acts as x-axis rotation value but with opposite signs of the actual x-rotation value.

`model.rotation = Rotation(x = 40.0f, y = 90.0f, z = 40.0f)`
only rotated around the y-axis by 90°

`model.rotation = Rotation(x = 0.0f, y = 90.0f, z = 40.0f)`
rotated around y-axis by 90° and front of car facing up (rotation around world's z-axis but not the model's z-axis)

This last example also meets the expectations
`model.rotation = Rotation(x = 40.0f, y = 90.0f, z = 0.0f)`
rotated around y-axis by 90° and front of car facing down (as expected because it's rotated around the model's x-axis)

**Comment by waxmar** [comment]
it looks like a Gimbal Effect and that's why Quaternions exist
have a look to that great video about Euler Angles issues in 3D to be sure that your case is not a normal behavior
https://youtu.be/zc8b2Jo7mno
Quaternions might seems difficult to start with but in fact, all you have to remember is that they are per axes rotation percentage expressed between -1.0 to 1.0 (x, y, z)
And a w value you can increase and decrease to affect a percentage to both x, y, z. 
> answer copied from Thomas Gorisse (discord chat)

**Comment by ThomasGorisse** [comment]
Thanks.

**Comment by ThomasGorisse** [comment]
In v4.0.0-rc.1, node rotation is set directly:

```kotlin
ModelNode(
    modelInstance = model,
    rotation = Rotation(x = 0f, y = 0f, z = 45f) // 45 degrees around Z
)
```

If you experienced Z-axis rotation issues in older versions, try upgrading to v4.0.0-rc.1. The transform system has been improved. Note that SceneView uses Filament's coordinate system (Y-up, right-handed).

**Comment by ThomasGorisse** [comment]
In v4, rotation uses Euler angles (degrees) via `Rotation` or quaternions via `Quaternion`:

```kotlin
ModelNode(
    modelInstance = model,
    rotation = Rotation(x = 0f, y = 45f, z = 90f)  // degrees
)
```

Or animate with `NodeAnimator`:
```kotlin
node.animateRotations(
    Rotation(0f, 0f, 0f),
    Rotation(0f, 0f, 360f)
).setDuration(2000).start()
```

Note: `animateRotations` was fixed in v4.0.0-rc.1 (#388) — the animated values now correctly propagate back to the node transform.
